src/dom-to-image-more.js

```diff
@@ -257,24 +258,38 @@
         return Promise.resolve(node)
             .then(makeNodeCopy)
             .then(function (clone) {
-                return cloneChildren(node, clone);
+                return cloneChildren(util.hasOpenShadowRoot(node) ? node.shadowRoot : node, clone);
             })
             .then(function (clone) {
                 return processClone(node, clone);
             });
 
         function makeNodeCopy(original) {
-            return util.isHTMLCanvasElement(original)
-                ? util.makeImage(original.toDataURL())
-                : original.cloneNode(false);
+            if (util.isHTMLCanvasElement(original)) {
+                return util.makeImage(original.toDataURL());
+            }
+            if (util.isInShadowRoot(original)) {
+                return cloneShadowNode(original);
+            }
+            return original.cloneNode(false);
+        }
+
+        function cloneShadowNode(original) {
+            const nextSibling = original.nextSibling;
+            const parentNode = original.parentNode;
+            // Bypass element encapsulation, by detaching from shadow node then restoring.
+            parentNode.removeChild(original);
```

Comment:
This seems like we're changing the original DOM, which is something we've never intentionally done before. Would this be a problem? Are we changing anything that the code/DOM on the clonee side would care about?

Reply:
*   We can't clone from the original shadow DOM unfortunately (`cloneNode` errors out).
*   This isn't that impactful to the end user - the change is reversed and the `Node` is restored.
*   We will be triggering MutationObserver callbacks by doing this.  
    I believe this might be an issue with SPAs that render views for a model collection in an app.  
    (Thinking lists or outlines e.g. chat users in a channel list, code symbols in VS Code's "Outline".)  
    However, if those apps are not taking ownership of those callbacks and shelling out to MutationObserver, that's arguably a poor design decision on their part.

Reply:
Turns out the logic to detach from the shadow DOM is only ever needed if you are trying to run `cloneNode` method on a `#shadow-root` fragment. I'll remove the logic sharpish.

Reply:
Resolved in d656297